auth/env_keychain.go

```diff
@@ -156,6 +156,10 @@ func authConfigToHeader(config *authn.AuthConfig) (string, error) {
 		return fmt.Sprintf("Basic %s", encoded), nil
 	}
 
+	if config.IdentityToken != "" {
+		return fmt.Sprintf("Bearer %s", config.IdentityToken), nil
```

Comment:
Looking at the [ggcr implementation](https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/transport/bearer.go#L124) I fear it is more complicated than this. We can't simply replace the registry token with an identity token but instead need to do an oauth flow where we use the identity token to get a refresh token.

Luckily I think the solution is to do less rather than to do more. When we store the resolved auth we [convert the AuthConfig to a http header](https://github.com/buildpacks/lifecycle/blob/main/auth/env_keychain.go#L121) and then [convert it back to an AuthConfig](https://github.com/buildpacks/lifecycle/blob/main/auth/env_keychain.go#L60). We could just not do that and return the original auth config - letting ggcr handle the complexity

I am can't remember why do this. It's either
1. historical reasons, we adapted existing code to changes in the ggcr interface to fix complation failures without taking advantage of improvements
2. some of this weirdness exists to support the `CNB_REGISTRY_AUTH` weirdness

In either case we should be able to make this improvement. We should just check what if any of the functionality pack uses and ensure it has what it needs.

Reply:
Thanks @ekcasey for the explanation. 

AFAIK pack doesn't care about the format of `CNB_REGISTRY_AUTH` (as long as it is something the lifecycle can understand). It seems like we are using the headers solely to "stringify" the `AuthConfig`, since we never return them when resolving the keychain.

WDYT about fe91721a - it's a bit hacky, but I'm hesitant to do a full-blown refactor of this interface lest we run into compatibility issues between pack <-> lifecycle.

Reply:
276e3dc4 removes the authConfig -> header -> authConfig round tripping when we are given a proper keychain (vs. an "env" one).